gatsby-node.js

```diff
@@ -9,15 +11,52 @@ const CAREER_DETAILS_TEMPLATE_PATH = path.resolve(
   'src/templates/career-details.tsx',
 );
 
+const { siteMetadata } = require('./gatsby-config');
 const PERSONIO_JOBS_URL = 'https://satellytes.jobs.personio.de/xml';
 const PERSONIO_SHORT_DESCRIPTION_NAME = 'Kurzbeschreibung';
 
+exports.onCreateNode = ({ node, getNode, actions, graphql }, options) => {
+  const { createNodeField } = actions;
+
+  if (node.internal.type === 'MarkdownRemark') {
+    const post = node.frontmatter;
+
+    // we need the title in order to generate anything
+    if (!post.title) {
+      return;
+    }
+
+    // we create a file name from the path (which is the page slug and more unique) but if none is available take the title
+    const fileName = `social-card---${slugify(post.path || post.title, {
+      lower: true,
+    })}.jpg`;
+    const outputFile = path.join('public', fileName);
+    const publicUrl = `${siteMetadata.siteUrl}${fileName}`;
+
+    generateCard({ title: post.title }, outputFile).then((filename) => {
+      createNodeField({
+        node,
+        name: `socialCard`,
+        value: publicUrl,
+      });
+    });
+  }
+};
+
```

Comment:
I think this approach is really beautiful. I stole it from an existing plugin:
https://github.com/andrioid/gatsby-plugin-social-cards/blob/0a18b30f50041b09721c3839c5c5d377781ecab6/gatsby-node.js#L10

We can't use the plugin as we have special requirements (two fonts for example) and a much simpler drawing solution (they use jimp we use node-canvas).

Reply:
Nice usage of gatsby nodes, really good 👍